File: common/src/main/java/com/hartwig/actin/datamodel/clinical/PriorSecondPrimary.java
package com.hartwig.actin.datamodel.clinical;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PriorSecondPrimary {

    @NotNull
    public abstract String tumorLocation();

    @NotNull
    public abstract String tumorSubLocation();

    @NotNull
    public abstract String tumorType();

    @NotNull
    public abstract String tumorSubType();

    @NotNull
    public abstract Set<String> doids();

    @Nullable
    public abstract Integer diagnosedYear();

    public abstract boolean isSecondPrimaryActive();

}

File: common/src/main/java/com/hartwig/actin/clinical/datamodel/PriorSecondPrimary.kt
package com.hartwig.actin.clinical.datamodel

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class PriorSecondPrimary {
    abstract fun tumorLocation(): String
    abstract fun tumorSubLocation(): String
    abstract fun tumorType(): String
    abstract fun tumorSubType(): String
    abstract fun doids(): Set<String?>
    abstract fun diagnosedYear(): Int?
    abstract fun diagnosedMonth(): Int?
    abstract fun treatmentHistory(): String
    abstract fun lastTreatmentYear(): Int?
    abstract fun lastTreatmentMonth(): Int?
    abstract fun status(): TumorStatus
}
